### DISCUSSION TITLE
pagination bug

### DISCUSSION THREAD

**Discussion by zcjwin** [encode/django-rest-framework#9038] [General] [answered]
## Checklist
Incoming page_size not effective

The page should be added on line 178 of the pagination. py file 'page_size_query_param = None' changed to page_size_query_param = 'page_size'

**Comment by auvipy** [comment]
can you please share more detail?

**Comment by shourav9884** [comment, marked as answer]
@zcjwin, it is described in the documentation that the default value of `page_size_query_param` is  `None` so that the client may not control the requested page size.

> `page_size_query_param` - If set, this is a string value indicating the name of a query parameter that allows the client to set the page size on a per-request basis. Defaults to `None`, indicating that the client may not control the requested page size.

ref: [Official documentation](https://www.django-rest-framework.org/api-guide/pagination/#configuration)

But you can set `PAGE_SIZE` globally in your settings. Example:
```python
REST_FRAMEWORK = {
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
    'PAGE_SIZE': 100
}
```

Even you can modify this behavior of `PageNumberPagination` by creating your own Pagination class which will override `PageNumberPagination` class. Example:
```python
class MyPageNumberPagination(PageNumberPagination):
    page_size = 100
    page_size_query_param = 'page_size'
    max_page_size = 1000
```
then use it in your View class like this
```python
class MyView(generics.ListAPIView):
    ...
    pagination_class = MyPageNumberPagination
```
Even you can use this custom pagination class globally 
```python
REST_FRAMEWORK = {
    'DEFAULT_PAGINATION_CLASS': 'apps.core.pagination.MyPageNumberPagination'
}    
```
Ref: [Modifying the pagination style](https://www.django-rest-framework.org/api-guide/pagination/#modifying-the-pagination-style)
